### ISSUE TITLE
How to finetune z-image-turbo?

### ISSUE THREAD

**Issue by hdjsjyl** [Tongyi-MAI/Z-Image#5]
Hi Author,
    Thanks for sharing excellent work.
    I would like to ask how to finetune the z-image-turbo model? Is there any reference code?
   Thank you.

Best

**Comment by bghira** [comment]
the way the paper makes it sound, you can introduce new information during distillation via reward signal. but it doesn't make a single mention to finetuning or any strategies one can take to do so.

sometimes you use part of the transformer (like SD3 Turbo paper) for use as GAN discriminator (LADD) but they don't describe for Z-Image Turbo (unless i missed it) how they created their discriminators, nor do they release the weights.

we could attempt to approximate finetuning of the model if you do what @jaretburkett did for Flux.1 Schnell by extensively finetuning a LoRA on Z-Image Turbo's own outputs which can then be used as an "assistant" adapter during finetuning of the Z-Image Turbo model. when you perform inference on the resulting weights, you shouldn't need the assistant adapter - its purpose is to "contain" the breakage during training time only.

but from what I've heard (I have no direct firsthand experience with it) that is not a 100% solution, it's more of a "it kinda works" trick.

without DMD loss, it's almost certain we'll break the distillation and destabilise the model.

**Comment by jjhaggar** [comment]
If you are fine with training a LoRA, juntaosun has written a mini-guide: https://github.com/Tongyi-MAI/Z-Image/issues/36#issue-3679005336
